File: addons/guide-cpp/src/modifiers/guide_modifier.cpp
#include "guide_modifier.h"

using namespace godot;

void GUIDEModifier::_bind_methods() {
    ClassDB::bind_method(D_METHOD("_editor_name"), &GUIDEModifier::_editor_name);
    ClassDB::bind_method(D_METHOD("_editor_description"), &GUIDEModifier::_editor_description);
}

GUIDEModifier::GUIDEModifier() {
}

GUIDEModifier::~GUIDEModifier() {
}

bool GUIDEModifier::is_same_as(const Ref<GUIDEModifier> &other) const {
    return this == other.ptr();
}

void GUIDEModifier::_begin_usage() {
}

void GUIDEModifier::_end_usage() {
}

Vector3 GUIDEModifier::_modify_input(Vector3 input, double delta, GUIDEAction::GUIDEActionValueType value_type) const {
    return input;
}

String GUIDEModifier::_editor_name() const {
    return "";
}

String GUIDEModifier::_editor_description() const {
    return "";
}

bool GUIDEModifier::_needs_physics_process() const {
    return false;
}

void GUIDEModifier::_physics_process(double delta) {
}

File: addons/GuideCS/Modifiers/GUIDEModifier.cs
using System;
using Godot;

namespace Guide;

[Tool]
[GlobalClass, Icon("res://addons/GuideCS/Modifiers/GUIDEModifier.svg")]
public partial class GUIDEModifier : Resource, IEditorDescription {
	public virtual bool IsSameAs(GUIDEModifier other) => this == other;

	public virtual void _BeginUsage() { }

	public virtual void _EndUsage() { }

	public virtual Vector3 _ModifyInput(Vector3 input, float delta, GUIDEAction.GUIDEActionValueType valueType) => input;

	public virtual string _EditorName() => "";

	public virtual string _EditorDescription() => "";

	public virtual bool _NeedsPhysicsProcess() => false;

	public virtual void _PhysicsProcess(float delta) { }
}
